src/core/include/openvino/pass/graph_rewrite.hpp

```diff
@@ -17,6 +17,52 @@ using graph_rewrite_callback = std::function<bool(pass::pattern::Matcher& m)>;
 using recurrent_graph_rewrite_callback = std::function<bool(pass::pattern::RecurrentMatcher& m)>;
 using handler_callback = std::function<bool(const std::shared_ptr<Node>& node)>;
 namespace pass {
+/// \brief Register openvino node pointers into container.
+/// Can create and/or add existing node pointers into register
+class NodeRegistry {
```

Comment:
Why did you move this class from developer to public API?

Reply:
To re-use it `graph_rewrite.hpp` header.
Developer API is not visible in this header.